### PR TITLE
chore(xmlupload): improve parse xml function

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**
Currently for the validate xml, the comments were not removed. This caused problems when finding the `encodings` attributes. However, we need the namespaces for the first validation step. Therefore a function was created that only parses the file and removes the comments.

### CODE REVIEW

**Comment by Nora-Olivia-Ammann** [comment]
> Now we have 2 functions that almost do the same:
> 
>     * `parse_and_remove_comments_xml_file()`
> 
>     * `parse_and_clean_xml_file()`
> 
> 
> This is confusing, because it is almost duplicating things, but with slight deviations. This makes it very hard to understand the paths that the program can take.
> 
> These code paths should be disentangled.

I agree, the reason why I did not do this here is because it opens a typing can of worms. If you would like to combine the functions, I am happy to do that, but I suggest another PR for that.

**Comment by jnussbaum** [comment]
> > Now we have 2 functions that almost do the same:
> > ```
> > * `parse_and_remove_comments_xml_file()`
> > 
> > * `parse_and_clean_xml_file()`
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is confusing, because it is almost duplicating things, but with slight deviations. This makes it very hard to understand the paths that the program can take.
> > These code paths should be disentangled.
> 
> I agree, the reason why I did not do this here is because it opens a typing can of worms. If you would like to combine the functions, I am happy to do that, but I suggest another PR for that.

Why would you like to do it in another PR? AFAIU, you introduced the duplication in the present PR. It doesn't seem natural to me to introduce a duplication/bad architecture in one PR, and then schedule a follow-up PR...?

**Comment by Nora-Olivia-Ammann** [comment]
> > > Now we have 2 functions that almost do the same:
> > > ```
> > > * `parse_and_remove_comments_xml_file()`
> > > 
> > > * `parse_and_clean_xml_file()`
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This is confusing, because it is almost duplicating things, but with slight deviations. This makes it very hard to understand the paths that the program can take.
> > > These code paths should be disentangled.
> > 
> > 
> > I agree, the reason why I did not do this here is because it opens a typing can of worms. If you would like to combine the functions, I am happy to do that, but I suggest another PR for that.
> 
> Why would you like to do it in another PR? AFAIU, you introduced the duplication in the present PR. It doesn't seem natural to me to introduce a duplication/bad architecture in one PR, and then schedule a follow-up PR...?

Because I wanted to focus on PR that introduce the new feature and this stood in the way. But if it is a deal-breaker, then I can do it here.

File: src/dsp_tools/utils/xml_utils.py
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/utils/xml_utils.py:1]: @jnussbaum 
I now added the suggested changes, if we wanted to be very consistent we should go through the repo and replace all the parse functions with one of these.

However I have a concern about the changes. The new function `parse_and_remove_comments_from_xml_file` returns the root. The function `_remove_qnames_and_transform_special_tags` uses the `.iter()` function. This function is not quite the same on a `etree._ElementTree[etree._Element]` (https://lxml.de/apidoc/lxml.etree.html#lxml.etree._ElementTree.iter) and on a `etree._Element` (https://lxml.de/apidoc/lxml.etree.html#lxml.etree._Element.iter). The main difference is that the `etree._Element.iter()` only does first depth iteration. While `etree._ElementTree.iter()` loops over all elements. The tests do not fail, but this difference still makes me hesitant to implement the change.
We could omit the `getroot()` but I don't like that option either as this is what we want in most places.

Thoughts?

Comment (jnussbaum) [reply src/dsp_tools/utils/xml_utils.py:1]: As far as I understand the docs, both `iter()` functions do exactly the same. The only difference is the wording of the docs, but if I look at the meaning behind the wording, I don't see a difference:

- ElementTree: "The iterator loops over all elements in this tree, in document order."
- Element: "Iterate over all elements in the subtree in document order (depth first pre-order), starting with this element."

Both iterate over all elements in document order.

**Review by jnussbaum — APPROVED** [review]

File: src/dsp_tools/utils/xml_utils.py
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/utils/xml_utils.py:1]: You are right, I misunderstood the wording.

**Pull request merged**